integrations/pooltogether-v4/components/form-yield-source-prize-pool-deposit.tsx

```diff
@@ -0,0 +1,157 @@
+  const prizePoolAddress = useLoadContractFromChainId(PRIZE_POOL_CONTRACT)
+  const usdcAddress = useLoadContractFromChainId(USDC_CONTRACT)
+
+  const { data: decimals } = useErc20Decimals({ address: usdcAddress })
+  const POWER = decimals != undefined ? BigNumber.from(10).pow(decimals) : BigNumber.from(10).pow(6)
+
+  const userBalance = useUserBalanceDeposit()
+  const isApproved = useUsdcApproval(userBalance)
+
+  const [depositAmount, setDepositAmount] = useState<number>()
+  const debouncedDepositAmount = useDebounce(BigNumber.from(depositAmount != undefined ? depositAmount * POWER.toNumber() : 0), 500)
+
+  const { data, write: depositToken } = usePoolTogetherPrizePoolDepositTo({
+    mode: 'recklesslyUnprepared',
+    address: prizePoolAddress,
+    args: [address || '0x0', BigNumber.from(debouncedDepositAmount)],
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading } = useWaitForTransaction({
+    hash: data?.hash,
+  })
+
+  const { data: approveData, write: approval } = useErc20Approve({
+    mode: 'recklesslyUnprepared',
+    address: usdcAddress,
+    args: [prizePoolAddress, approvalAmount],
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading: loadApprove, isSuccess: successApprove } = useWaitForTransaction({
+    hash: approveData?.hash,
+  })
+
+  useEffect(() => {
+    isChecked ? setApprovalAmount(ethers.constants.MaxInt256) : setApprovalAmount(BigNumber.from(debouncedDepositAmount))
+  }, [isChecked])
+
+  useEffect(() => {
+    if (successApprove || submitDeposit) {
+      depositToken?.()
+      setSubmitDeposit(false)
+    }
+  }, [successApprove, submitDeposit])
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    if (depositAmount != undefined && depositAmount >= 2.0) {
+      if (!isApproved) {
+        approval?.()
+      } else {
+        setSubmitDeposit(true)
+      }
+    } else {
+      setValidAmount(false)
+    }
+  }
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value != '' ? parseFloat(event.target.valueAsNumber.toFixed(decimals)) : undefined
+    console.log(value)
+    setDepositAmount(value)
+    setApprovalAmount(value != undefined ? BigNumber.from(value * POWER.toNumber()) : BigNumber.from(0))
+  }
+
+  const handleAmount = () => {
+    setDepositAmount(userBalance)
+    setApprovalAmount(BigNumber.from(userBalance * POWER.toNumber()))
+  }
+
+  return (
+    <>
+      <Form.Root onSubmit={handleSubmit}>
+        <Form.Field name="amountDeposit">
+          <div className="flex justify-between align-baseline">
+            <Form.Label className="FormLabel mb-2">Amount</Form.Label>
+            <Form.Label className="FormLabel mb-2">
+              <a className="ml-10 cursor-pointer hover:underline" onClick={() => handleAmount()}>
+                {parseFloat(userBalance.toString()).toFixed(2)} USDC
+              </a>
```

Comment:
You shoudn't be using an anchor tag for that, it's suited for links only, I suggest changing it to a `span`
```suggestion
              <span className="ml-10 cursor-pointer hover:underline" onClick={() => handleAmount()}>
                {parseFloat(userBalance.toString()).toFixed(2)} USDC
              </span>
```